package large_coref.protobuf;

option java_package = "cc.refectorie.proj.relation.protobuf";
option java_outer_classname = "DocumentProtos";

message Document {
  required string filename = 1;
  repeated Sentence sentences = 2;

  message Sentence {
    repeated Token tokens = 1;
    repeated Mention mentions = 2;
    optional DepTree depTree = 3;
  }

  message Token {
    required string word = 1;
    optional string tag = 2;
    optional string ner = 3;
  }

  message Mention {
    required int32 id = 1;
    optional string entityGuid = 2;

    required int32 from = 3;
    required int32 to = 4;
    required string label = 5;

  }

  message DepTree {
    required int32 root = 1;
    repeated int32 head = 2;
    repeated string relType = 3;
  }

  message RelationMention {
    required int32 id = 1;
    required int32 source = 2;
    required int32 dest = 3;
    required string label = 4;
  }


}

message Relation {
  required string sourceGuid = 1;
  required string destGuid = 2;
  required string relType = 3;
  repeated RelationMentionRef mention = 4;

  message RelationMentionRef {
    required string filename = 1;
    required int32 sourceId = 2;
    required int32 destId = 3;
    repeated string feature = 4;
    optional string sentence = 5;
  }
}

message Entity {
  required string guid = 1;
  optional string name = 2;
  optional string type = 3;
  optional string pred = 4;
  repeated EntityMentionRef mention = 5;

  message EntityMentionRef {
    required string filename = 1;
    required int32 id = 2;
    repeated string feature = 3;
  }

}





